### DISCUSSION TITLE
Parameter should be a read only type: what am I doing wrong?

### DISCUSSION THREAD

**Discussion by mward-sudo** [jonaskello/eslint-plugin-functional#378] [Q&A] [answered]
```
type props = {
  readonly author: PostAuthorFragment
  readonly date: string
  readonly authorImgProps: CloudinaryImageProps | null
}

export const AuthorDetails = ({ author, date, authorImgProps }: props) => {
  return (
    ...
  )
}
```

I'm getting the Parameter should be a read only type error and I cannot figure what I am doing incorrectly.

Can anyone help?

**Comment by RebeccaStevens** [comment, marked as answer]
Is it this rule repotting the error: [`@typescript-eslint/prefer-readonly-parameter-types`](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/prefer-readonly-parameter-types.md)?

This rule checks typed deeply for readonlyness. So if `PostAuthorFragment` or `CloudinaryImageProps` isn't deeply readonly the an issue will be reported`.

**Reply by mward-sudo** [reply]
Yes, it is that rule.

I've changed graphql code generator to generate immutable types, but I'm still getting the error.

This is the full code for the types involved:

```
type PostAuthorFragment = {
  readonly __typename?: 'graphcms_Author'
  readonly name: string
  readonly twitterHandle?: string | null
  readonly picture?: {
    readonly __typename?: 'graphcms_Asset'
    readonly handle: string
    readonly height?: number | null
    readonly width?: number | null
  } | null
}

type CloudinaryImageProps = {
  readonly src: string
  readonly srcSet: string
  readonly sizes: string
  readonly alt: string
  readonly width: number
  readonly height: number
  readonly style: {
    readonly backgroundImage: string
    readonly backgroundSize: string
    readonly aspectRatio: string
  }
}

type props = {
  readonly author: PostAuthorFragment
  readonly date: string
  readonly authorImgProps: CloudinaryImageProps | null
}

export const AuthorDetails = ({ author, date, authorImgProps }: props) => {
  return (
    ...
  )
}
```

But I still get the error at `({ author, date, authorImgProps }: props)`

**Reply by mward-sudo** [reply]
I reloaded vscode and the error now seems to have gone away - maybe there was some caching involved?
